.github/workflows/rust.yml

```diff
@@ -74,7 +74,7 @@ jobs:
       matrix:
         os: [ ubuntu-18.04, macos-10.15 ]
         # See rust-toolchain for why we're using nightly here.
-        toolchain: [ nightly-2021-09-03 ]
+        toolchain: [ nightly-2021-11-24 ]
```

Comment:
what would happen if we didn't specify this?

Reply:
The short answer is that I'm not sure and I don't remember the history here.

In principle, this approach makes it easy for us to test with a few different toolchains (e.g., nightly, stable, beta).  I'm not sure how that works with the rust-toolchain file in the repo.  But [the docs imply that the input here takes precedence](https://github.com/actions-rs/toolchain#the-toolchain-file) -- I'm not sure how they accomplish that.

If we didn't specify anything at all for the toolchain, I would have expected that the build would be invoked with the latest stable Rust, find the rust-toolchain file, download and install the one specified there, and then use it.  Builds would take a bit longer, though that's not necessarily a good reason for this redundancy.  And now that I read that sentence from the docs, I'm not even sure if that's true -- maybe the action looks at the rust-toolchain file before deciding on which toolchain to install in the first place.